nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -650,6 +650,28 @@ Costmap2DROS::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameter
         if (makeFootprintFromString(footprint_, new_footprint)) {
           setRobotFootprint(new_footprint);
         }
+      } else if (name == "robot_base_frame") {
+        // First, make sure that the transform between the robot base frame
+        // and the global frame is available
+        std::string tf_error;
+
+        RCLCPP_INFO(get_logger(), "Checking transform");
+        rclcpp::Rate r(2);
+        while (rclcpp::ok() &&
+          !tf_buffer_->canTransform(
```

Comment:
This concerns me a little. I totally agree that this is needed, but in the situation that it can't transform because of a typo, does this lock up the system really badly if the node is spinning on this callback for awhile? 

I'd be concerned about safety, this seems almost like something we may want to have only changable in a completely transitioned inactive state. 

I could be fine with this, but I would like you to characterize what happens when this callback goes wrong so we know the symptoms

Reply:
We should definitely discuss this point and the desired behavior! I guess in its current state it is blocking indefinitely if there is a typo.
I propose to check the transform and if it is not available after a certain timeout (to discuss but it should be short), reject the parameter change (`result.successful = false;`) and leave the original value of the param `robot_base_frame` unchanged. 
What do you think ?

Reply:
That seems very reasonable to me. Maybe give it 1 full second to make it work?

Reply:
Yes!